src/main/java/com/rabbitmq/http/client/Client.java

```diff
@@ -371,6 +373,44 @@ public void closeConnection(String name, String reason) {
     deleteIgnoring404(uri, headers);
   }
 
+  /**
+   * Retrieves state and metrics information for all consumers across the cluster.
+   *
+   * @return list of consumers across the cluster
+   */
+  public List<ConsumerDetails> getConsumers() {
+    final URI uri = uriWithPath("./consumers/");
+    return Arrays.asList(this.rt.getForObject(uri, ConsumerDetails[].class));
+  }
+
+
+  /**
+   * Retrieves state and metrics information for all consumers across the cluster.
+   * using query parameters
+   *
+   * @param queryParameters
+   * @return list of consumers across the cluster
+   */
+  @SuppressWarnings("unchecked")
+  public Page<ConsumerDetails> getConsumers(QueryParameters queryParameters) {
+    final URI uri = uriWithPath("./consumers/", queryParameters);
+    ParameterizedTypeReference<Page<ConsumerDetails>> type = new ParameterizedTypeReference<Page<ConsumerDetails>>() {
+    };
+    return (queryParameters.pagination().hasAny()) ? this.rt.exchange(uri, HttpMethod.GET, null, type).getBody() :
+            new Page(this.rt.getForObject(uri, ChannelInfo[].class));
+  }
+
+  /**
+   * Retrieves state and metrics information for individual consumer.
+   *
+   * @param name channel name
+   * @return consumer information
+   */
+  public ConsumerDetails getConsumer(String name) {
+    final URI uri = uriWithPath("./consumer/" + encodePathSegment(name));
```

Comment:
I don't see this endpoint in the [HTTP API dispatcher](https://github.com/rabbitmq/rabbitmq-server/blob/master/deps/rabbitmq_management/src/rabbit_mgmt_dispatcher.erl#L120-L121).

If you mean `GET /api/consumers/{host}` then the docs are wrong as it most certainly does not return information about a specific consumer.

Here is a proof that `GET /consumer/{name}` does not exist but `GET /api/consumers/{host}` does:

```
curl -X GET -u guest:guest http://localhost:15672/api/consumers | jq
[
  {
    "arguments": {},
    "ack_required": false,
    "active": true,
    "activity_status": "up",
    "channel_details": {
      "connection_name": "127.0.0.1:60578 -> 127.0.0.1:5672",
      "name": "127.0.0.1:60578 -> 127.0.0.1:5672 (1)",
      "node": "rabbit@hostname-a02",
      "number": 1,
      "peer_host": "127.0.0.1",
      "peer_port": 60578,
      "user": "guest"
    },
    "consumer_tag": "bunny-1624904598000-845343168135",
    "exclusive": false,
    "prefetch_count": 0,
    "queue": {
      "name": "qq.10",
      "vhost": "/"
    }
  }
]


curl -X GET -u guest:guest http://localhost:15672/api/consumers/%2f | jq
[
  {
    "arguments": {},
    "ack_required": false,
    "active": true,
    "activity_status": "up",
    "channel_details": {
      "connection_name": "127.0.0.1:60578 -> 127.0.0.1:5672",
      "name": "127.0.0.1:60578 -> 127.0.0.1:5672 (1)",
      "node": "rabbit@hostname-a02",
      "number": 1,
      "peer_host": "127.0.0.1",
      "peer_port": 60578,
      "user": "guest"
    },
    "consumer_tag": "bunny-1624904598000-845343168135",
    "exclusive": false,
    "prefetch_count": 0,
    "queue": {
      "name": "qq.10",
      "vhost": "/"
    }
  }
]


curl -X GET -u guest:guest http://localhost:15672/api/consumer/bunny-1624904598000-845343168135 | jq
{
  "error": "Object Not Found",
  "reason": "Not Found"
}
```

Reply:
Yes, you are correct, copy and paste got me. Updating Documentation now!

Reply:
Wow I really missed that since it actually returns a List<ConsumerDetails>. I ended up only using the no argument version in my application, thus I never ran into this. I have committed back and will resolve conversation.